### PR TITLE
[scan] support closure

### PR DESCRIPTION
**Pull request by ydwu4**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #135603
* #137967
* __->__ #135602
* #135601
* #135600

This PR adds an additional_inputs argument to support closures similar to what we've done for while_loop.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

File: torch/_higher_order_ops/scan.py [hunk 1]
```diff
@@ -186,14 +188,16 @@ class ScanOp(HigherOrderOperator):
     def __init__(self):
         super().__init__("scan")
 
-    def __call__(self, combine_fn, init, xs, dim, reverse):
-        return super().__call__(combine_fn, init, xs, dim, reverse)
+    def __call__(self, combine_fn, init, xs, dim, reverse, additional_inputs):
+        return super().__call__(combine_fn, init, xs, dim, reverse, additional_inputs)
```
Comment (zou3519) [inline torch/_higher_order_ops/scan.py:192]: can we maintain an invariant that additional_inputs is always a Tuple?

Comment (zou3519) [reply torch/_higher_order_ops/scan.py:192]: Oh, do can these be symints? Yikes

Comment (zou3519) [reply torch/_higher_order_ops/scan.py:192]: I guess this comment is: "let's assert additional_inputs is a List" here.

File: test/functorch/test_control_flow.py [hunk 2]
```diff
@@ -2838,7 +2838,7 @@ def forward(self, fct_1, init_1, xs_1):
     clone = torch.ops.aten.clone.default(init_1);  clone = None
     select_copy = torch.ops.aten.select_copy.int(xs_1, 0, 0);  select_copy = None
     scan_combine_graph_0 = self.scan_combine_graph_0
-    scan = torch.ops.higher_order.scan(scan_combine_graph_0, [init_1], [xs_1], 0, True);  scan_combine_graph_0 = init_1 = xs_1 = None
+    scan = torch.ops.higher_order.scan(scan_combine_graph_0, [init_1], [xs_1], 0, True, []);  scan_combine_graph_0 = init_1 = xs_1 = None
```
Comment (zou3519) [inline test/functorch/test_control_flow.py:2841]: Btw, I have a slight preference for Tuple over List in general for all three of these lists. The thing about List is that it is mutable and that leads to footguns, if someone writes a scan_dense that mutates the list then we've got a problem.

My question here is: are we being consistent with what we use for the other control flow ops? Let's be consistent with everything. If we're not consistent, then let's pick one of the two to go with

**Review by zou3519 — APPROVED** [review]

File: test/functorch/test_control_flow.py (hunk 2 above)
Comment (ydwu4) [reply test/functorch/test_control_flow.py:2848]: It's a bit chaotic right now for cond and scan, it's a list but for while_loop it's a tuple. Let me add a TODO for unifying them.

**Pull request closed without merging**